ruby.c

```diff
@@ -1522,7 +1507,7 @@ proc_options(long argc, char **argv, ruby_cmdline_options_t *opt, int envopt)
 #endif
             }
             else if (strcmp("yjit", s) == 0 || setup_yjit_options(s, &opt->yjit)) {
-#if USE_MJIT
+#if YJIT_BUILD
```

Comment:
Looks like `setup_yjit_options()` is still called when `!YJIT_BUILD`. Maybe we can copy the else if branch for MJIT that's above this?